pyqtorch/circuit.py

```diff
@@ -64,15 +65,24 @@ def forward(self, state: State, values: dict[str, Tensor] = {}) -> State:
     def device(self) -> torch_device:
         return self._device
 
+    @property
+    def dtype(self) -> dtype:
+        return self._dtype
+
     def init_state(self, batch_size: int = 1) -> Tensor:
-        return zero_state(self.n_qubits, batch_size, device=self.device)
+        return zero_state(self.n_qubits, batch_size, device=self.device, dtype=self.dtype)
 
     def reverse(self) -> QuantumCircuit:
         return QuantumCircuit(self.n_qubits, ModuleList(list(reversed(self.operations))))
 
-    def to(self, device: torch_device) -> QuantumCircuit:
-        self.operations = ModuleList([op.to(device) for op in self.operations])
-        self._device = device
+    def to(self, *args: Any, **kwargs: Any) -> QuantumCircuit:
+        self.operations = ModuleList([op.to(*args, **kwargs) for op in self.operations])
+        try:
+            self._device = self.operations[0].device
+            self._dtype = self.operations[0].dtype
+        except IndexError:
```

Comment:
Wouldn't it be preferrable to check for `self.operations` emptiness ?

Reply:
true